test/base/BaseModel.spec.js

```diff
@@ -4,10 +4,49 @@ import _isString from 'lodash/isString';
 import BaseModel from '../../src/assets/scripts/client/base/BaseModel';
 import ExtendedBaseModelFixture from './_fixtures/ExtendedBaseModelFixture';
 
+ava('makes sure the model in instantiates with any data type', t => {
+    t.notThrows(() => new BaseModel());
+    t.notThrows(() => new BaseModel([]));
+    t.notThrows(() => new BaseModel({}));
+    t.notThrows(() => new BaseModel(42));
+    t.notThrows(() => new BaseModel('string'));
+    t.notThrows(() => new BaseModel(false));
+});
+
 ava('instantiates with an _id property', t => {
-    const result = new BaseModel();
+    const resultEmpty = new BaseModel();
+
+    t.true(_isString(resultEmpty._id));
+});
+
+ava('instantiates with an _id with string property', t => {
+    const resultString = new BaseModel('string');
+
+    t.true(_isString(resultString._id));
+});
+
+ava('instantiates with an _id with an Object', t => {
+    const resultObj = new BaseModel({});
+
+    t.true(_isString(resultObj._id));
+});
+
+ava('instantiates with an _id with an array', t => {
+    const resultArray = new BaseModel([]);
```

Comment:
There are cases in the code where it is currently being called by other parameters. I was confused on why that was the case though.